tests/vmi_multus_test.go

```diff
@@ -864,10 +864,19 @@ var _ = Describe("[Serial]SRIOV", func() {
 			waitVmi(vmi1)
 			waitVmi(vmi2)
 
-			vmi1, err = virtClient.VirtualMachineInstance(vmi1.Namespace).Get(vmi1.Name, &metav1.GetOptions{})
-			Expect(err).NotTo(HaveOccurred())
-			vmi2, err = virtClient.VirtualMachineInstance(vmi2.Namespace).Get(vmi2.Name, &metav1.GetOptions{})
-			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() string {
```

Comment:
I fail to understand how this change is related to the production code change introduced in this PR.

I do expect to see in one of the tests a check about the network appearing in the VMI status report.

If you use `Eventually` to resolve the Guest Agent data update race, then I think there is already a dedicated PR for it and it should fit there.
If there is a need to wait for the data to get populated that is a per-test retry responsibility focused on what it checks and not part of the VMI creation.

Reply:
The PR #4511 you talk on is this i think.
But in this PR i added the feature of the name, so i added the e2e test that checks it,
this test didnt have getInterfaceNetworkNameByMAC logic, this is what i added, and i must wrap it with Eventually because of the race.

Reply:
Since i moved moved the Skip, 
and  now `createSriovVMs` does run, and invoke `getInterfaceNameByMAC` which depends on the VMI status,
the `Eventually` that i added is mandatory both to check the network name and to know that the VMI status is available, this is why i added it there.

I will remove from this PR that first commit that does the Skip change + refactor,
cherry-pick #4511 here, and add logic to 3985 to check the name exists as well,
this is the right thing imo, because once #4448 is merged, `getInterfaceNameByMAC` would be removed, and we dont want the network name check to be removed, and also indeed 3985 is more natural place for it.